Create and mint hats using the multisig

pheuberger opened the pull request:
This adds creating and minting multicall support for hats and allows the user to also send that through a Gnosis safe.

pheuberger committed:
fix: useHatsClient flakyness

The hats client was very temperamental. The publicClient dependency on
the useEffect hook wasn't working as expected, so it was replaced with
the walletClient, which is now also coming from its own hook.

pheuberger committed:
feat: send hat multicall using safe

This also checks if the user is on the safe and only allows them to
submit the form if they are.

bitbeckers reviewed (changes requested):
Fix merge conflixt then shiiiiiiip

pheuberger committed:
chore: use new hats tree

Also introduce a constants file.

pheuberger committed:
feat: create and mint hat

The hook exposes a wrapped client of the hats protocol with currently
one function: createAndMintHat().

This function uses the browser wallet to do a multicall of these two
transactions:

  - create a hat
  - mint the hat that was just created
